arclightning_backend/src/router.rs

```diff
@@ -143,4 +143,27 @@ impl Router {
             _ => self.invalid_endpoint(),
         }
     }
+
+   
 }
+
+fn list_files(dir: PathBuf) -> Result<Vec<PathBuf>, io::Error>{
+    
+    if dir.is_dir(){
```

Comment:
Instead of assigning `paths` and using `Ok()` to return paths in both branches of the `if/else`, you could try the following:

`let result = if dir.is_dir() { . . .} else { . . .}`
`Ok(result)`